repl/src/main/resources/fake_shell.py

```diff
@@ -426,11 +426,14 @@ def magic_table(name):
                 }
                 headers[name] = header
             else:
-                # Reject columns that have a different type.
-                if header['type'] != col_type:
-                    exc_type = Exception
-                    exc_value = 'table rows have different types'
-                    return execute_reply_error(exc_type, exc_value, None)
+                # Reject columns that have a different type. (allow none value)
+                if col_type is not "NULL_TYPE" and header['type'] != col_type:
```

Comment:
change "is not" -> "!=" 
this is exception case : if header column type is not null, and column type different type to header.

if i'm wrong understand your question. comment again plaese.